autoload/coc/powershell.vim

```diff
@@ -1,18 +1,34 @@
 let s:root           = expand('<sfile>:h:h:h')
 let s:is_win         = has('win32') || has('win64')
-let s:is_mac = !s:is_win && !has('win32unix')
+let s:is_mac         = !s:is_win && !has('win32unix')
       \ && (has('mac') || has('macunix') || has('gui_macvim') ||
       \   (!isdirectory('/proc') && executable('sw_vers')))
 let s:is_vim         = !has('nvim')
-let s:install_script = s:root.'/'.(s:is_win ? 'install.cmd' : 'install.ps1')
+let s:install_script = s:root.'/install.ps1'
 
-if(s:is_mac)
-    let s:install_script = 'pwsh '.s:install_script
-endif
+function! coc#powershell#install(...)
+    if(exists('a:1.flags'))
+        let s:flags = a:1.flags
+    else
+	let s:flags = ''
+    endif
+    let s:powershell_executable = ''
+    if(exists('a:1.powershellExecutable'))
+        let s:powershell_executable = a:1.powershellExecutable
+    else
+        if(s:is_mac)
```

Comment:
I did this because the install script has a shebang which works on linux... but maybe we should just get rid of the if (so `pwsh` is also added on Linux) so that we can use `-NoProfile` for the perf gain.

Reply:
I took out the check for Mac, if it's not specified in the call to the function, we set it to pwsh and then only change it if we're on Windows. Also added `-NoProfile` and `-ExecutionPolicy Bypass` to the call to PowerShell.

Also added `-File` because I didn't think it was working on Linux, turns out there was no space, so that might be unneeded.

I've "tested" this on both Linux and Windows.